### ISSUE TITLE
Some questions about the CUB dataset

### ISSUE THREAD

**Issue by judgment404** [mashijie1028/Happy-CGCD#1]
Thanks for your work! I attempted to train on CUB following the instructions in the README.
In stage-0, the last few lines of log were:

```
2024-11-07 01:51:07.908 | INFO     | __main__:train_offline:109 - Epoch: [98][0/18]	 loss 3.90938	 cls_loss: 0.1097 cluster_loss: 0.8087 sup_con_loss: 1.0805 contrastive_loss: 4.5649 
2024-11-07 01:51:15.254 | INFO     | __main__:train_offline:109 - Epoch: [98][10/18]	 loss 3.83976	 cls_loss: 0.1063 cluster_loss: 0.7423 sup_con_loss: 1.0133 contrastive_loss: 4.5622 
2024-11-07 01:51:20.481 | INFO     | __main__:train_offline:112 - Train Epoch: 98 Avg Loss: 3.8567 
2024-11-07 01:51:20.482 | INFO     | __main__:train_offline:114 - Testing on disjoint test set...
2024-11-07 01:51:30.236 | INFO     | __main__:train_offline:116 - Test Accuracies: All 0.8530 | Old 0.8530
2024-11-07 01:51:31.018 | INFO     | __main__:train_offline:128 - model saved to /home/gentoo/code/Happy-CGCD/dev_outputs_Happy_offline/Happy/cub/Old100_Ratio0.8_20241107-010856/checkpoints/model.pt.
2024-11-07 01:51:31.019 | INFO     | __main__:train_offline:139 - Exp Name: happy-offline
2024-11-07 01:51:31.019 | INFO     | __main__:train_offline:140 - Metrics with best model on test set: Old: 0.8565
2024-11-07 01:51:31.019 | INFO     | __main__:train_offline:141 - 

2024-11-07 01:51:33.653 | INFO     | __main__:train_offline:109 - Epoch: [99][0/18]	 loss 3.92888	 cls_loss: 0.1090 cluster_loss: 0.8233 sup_con_loss: 1.1090 contrastive_loss: 4.5653 
2024-11-07 01:51:40.984 | INFO     | __main__:train_offline:109 - Epoch: [99][10/18]	 loss 3.87933	 cls_loss: 0.1076 cluster_loss: 0.7726 sup_con_loss: 1.0693 contrastive_loss: 4.5618 
2024-11-07 01:51:46.194 | INFO     | __main__:train_offline:112 - Train Epoch: 99 Avg Loss: 3.8638 
2024-11-07 01:51:46.199 | INFO     | __main__:train_offline:114 - Testing on disjoint test set...
2024-11-07 01:51:55.730 | INFO     | __main__:train_offline:116 - Test Accuracies: All 0.8530 | Old 0.8530
2024-11-07 01:51:56.530 | INFO     | __main__:train_offline:128 - model saved to /home/gentoo/code/Happy-CGCD/dev_outputs_Happy_offline/Happy/cub/Old100_Ratio0.8_20241107-010856/checkpoints/model.pt.
2024-11-07 01:51:56.530 | INFO     | __main__:train_offline:139 - Exp Name: happy-offline
2024-11-07 01:51:56.531 | INFO     | __main__:train_offline:140 - Metrics with best model on test set: Old: 0.8565
2024-11-07 01:51:56.531 | INFO     | __main__:train_offline:141 - 
```
And in stage-1, the log were:
```
2024-11-07 09:20:21.402 | INFO     | models.utils_proto_aug:update_prototypes_online:151 - computing (predicted) class-wise mean...
2024-11-07 09:20:21.402 | INFO     | models.utils_proto_aug:update_prototypes_online:151 - computing (predicted) class-wise mean...
2024-11-07 09:20:21.402 | INFO     | models.utils_proto_aug:update_prototypes_online:151 - computing (predicted) class-wise mean...
2024-11-07 09:20:21.405 | INFO     | __main__:<module>:732 - Saving ProtoAugDict to /home/gentoo/code/Happy-CGCD/dev_outputs_Happy_online/Happy/cub_shuffle_seed_0/Old100_Ratio0.8_ContinualNum5_UnseenNum25_SeenNum5_20241107-082203/checkpoints/ProtoAugDict_session-5.pt.
2024-11-07 09:20:21.407 | INFO     | __main__:<module>:745 - Saving results (best acc list) to /home/gentoo/code/Happy-CGCD/dev_outputs_Happy_online/Happy/cub_shuffle_seed_0/Old100_Ratio0.8_ContinualNum5_UnseenNum25_SeenNum5_20241107-082203/checkpoints/best_acc_list_session-5.pt.
2024-11-07 09:20:21.408 | INFO     | __main__:<module>:749 - 

==================== print final results over 5 continual sessions ====================
2024-11-07 09:20:21.408 | INFO     | __main__:<module>:751 - Session-1: All (Hard): 0.7656 Old: 0.8394 New: 0.4055 | All (Soft): 0.7656 Seen: 0.8394 Unseen: 0.4055
2024-11-07 09:20:21.408 | INFO     | __main__:<module>:751 - Session-2: All (Hard): 0.6750 Old: 0.7916 New: 0.3926 | All (Soft): 0.6750 Seen: 0.7172 Unseen: 0.4303
2024-11-07 09:20:21.408 | INFO     | __main__:<module>:751 - Session-3: All (Hard): 0.6425 Old: 0.7943 New: 0.3936 | All (Soft): 0.6425 Seen: 0.6698 Unseen: 0.4470
2024-11-07 09:20:21.408 | INFO     | __main__:<module>:751 - Session-4: All (Hard): 0.5937 Old: 0.7909 New: 0.3527 | All (Soft): 0.5937 Seen: 0.6364 Unseen: 0.2626
2024-11-07 09:20:21.408 | INFO     | __main__:<module>:751 - Session-5: All (Hard): 0.5585 Old: 0.7971 New: 0.3253 | All (Soft): 0.5585 Seen: 0.5864 Unseen: 0.3112
```

But I got numbers (76, 67, 64, 59, 55) across all sessions lower than those in the paper (81, 74, 67, 62, 59). I trained using the default hyperparameters in README.md, and I’m not sure if I misconfigured the training parameters or something else. I think maybe there are something wrong in my experiment, do you have any idea can help me fix the problem?

I look forward to your response at your earliest convenience.

**Comment by mashijie1028** [comment]
Hi! Thanks for the question. It seems that you got the lower accuracy in Stage-0, *i.e.*, 85.65<90.26 (our paper), so the subsequent results in continual learning became worse, accordingly.

I have checked the args of the logs and make sure  `README.md` is consistent with our experiments. You should first re-run the experiments and make sure the results are aligned with our paper in **offline Stage-0**.

Besides, I share the offline ckpt and logs via [Baidu NetDisk](https://pan.baidu.com/s/1AFdxN8ZjUCTJ039uMGltVQ?pwd=cgcd), which are exactly the same as the reported results. You could also directly begin continual training using this ckpt. Hope this helps! 🍺
